### ISSUE TITLE
Ansible 2.8.0 bnreaks lots of things

### ISSUE THREAD

**Issue by baradhili** [roots/trellis#1085]
* [X ] I've read the [guidelines for Contributing to Roots Projects](https://github.com/roots/.github/blob/master/CONTRIBUTING.md)
* [ X] This request isn't a duplicate of an existing issue
* [ X] I've read the [docs](https://roots.io/documentation/) and followed them (if applicable)
* [ X] This is not a personal support request that should be posted on the [Roots Discourse](https://discourse.roots.io/) community

## Description

Ansible 2.8.0 breaks quite a few roles inc logrotate - this seems to be due to switching over to Python 3.0

Fix is either to lock your ansible to 2.7.x or to manually upgrade roles

**Comment by retlehs** [comment]
thanks for the issue! only 2.7.5 max is supported at the moment - we could do a better job at communicating this :D 

https://github.com/roots/trellis/blob/49cf5de36c7bfb0661a922288da898228cc526e9/vagrant.default.yml#L7

**Comment by nextgenthemes** [comment]
Why 2.7.5 specifically and what exactly breaks?

I actually setup my staging server a few days ago with the ansible version that comes with open suse tumbleweed. I thought the version was already 2.8.x but not 100% sure. I think it worked without any errors I know if, not sure if some things like logrotate do not work correctly on the server now did not check that, the ansible setup passed! But now that ansible has probably updated I get this:

> ERROR! Unexpected Exception, this is probably a bug: 'PlaybookCLI' object has no attribute 'options'

But even the not rolling release of OpenSUSE is already on 2.7.10 and the release b4 that has 2.7.8 https://software.opensuse.org/package/ansible

So I think its not about

> communicating this 

but actually supporting the latest versions! Its not possible to just easy downgrade a package on mayor Linux distros AFAIK so I am really not sure what to do right now.

**Comment by retlehs** [comment]
> but actually supporting the latest versions!

This is an open source project where anyone is free to contribute. We try to welcome it as much as we can. Right now there is not anyone that has expressed interest or has the free time to keep up with this. 

Want to help? That would be great and appreciated 🌻  Want to financially support our OSS projects so we can help get people paid to fix things like this? We have a Patreon 🌱 

Use pip to install Ansible and you can easily and quickly install any version you’d like... this is covered in the documentation and plenty of forum posts.

**Comment by baradhili** [comment]
yes, I only opened the issue so that the next person wondering why it stopped working would see something..

@nextgenthemes  to downgrade, remove ansible from your install and then manually install 2.7.x version - what changed is that ansible uses python 3.x now - so bucketloads of things are now broken under ansible 2.8 - mostly dependencies, I don't think there is actually any faults in Trellis.

**Comment by nextgenthemes** [comment]
I never said its anyone fault I know how things go, I just did not like the way the response sounded.

Sorry I was annoyed because I did not expect it to be easy to downgrade and did not even think about pip. So unless there are any security issues that come with saying on 2.7.5 I have no issue staying on a old version for now. But pip itself is actually a security concern [I recently learned. ](https://prahladyeri.com/blog/2019/06/package-signing-in-pip-it-works-in-a-roundabout-sort-of-way.html) but I just read on this after I already installed it without checking any sighing. Read the [old post](https://dev.to/prahladyeri/why-doesn-t-the-python-package-manager-pip-have-package-signing-13ll) that assumed there is non at all in pip and its pretty much like that because nobody is using it. Basically in pip a package can be taken over by someone and the user well never notice that someone got full control over a package while updating or installing, horrible! I think npm also does not have it.

In case this may help someone what I actually did was this:
```
sudo zypper remove ansible
pip install --user ansible==2.7.5
```
I then needed `~/.local/bin` in my path and it was not by default, its probably not on SUSE or its because I switched from bash to zsh.

I installed it for my user only just because that was the first command I found on the pip docs while searching for how to install a specific version, sounded better to me but no specific reason.

I guess I was on a version >2.7.5,<2.8 b4 and every version below 2.8.x actaully still works.

**Comment by chancestrickland** [comment]
Posting this here in case it's helpful to folks who are new to Trellis, run into versioning errors and now need to downgrade Ansible: If you installed via Homebrew, I spun up a tap to install 2.7 easily (pip was giving me some grief here).

```
brew uninstall ansible
brew install chancestrickland/ansible-2-7/ansible@2.7
```

**Comment by flocca** [comment]
> ```
> brew uninstall ansible
> brew install chancestrickland/ansible-2-7/ansible@2.75
> ```

This doesn't work for me.

This works and will install Ansible 2.7.5 using a previous formula from `homebrew-core`:

```
brew uninstall ansible
brew install https://raw.githubusercontent.com/Homebrew/homebrew-core/0b33ebc253ff9acb8d8365bd569e38d04c236f41/Formula/ansible.rb
```

**Comment by raulillana** [comment]
For those who have installed via pip:

> pip uninstall ansible
> pip install ansible==2.7.5
